src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/base/NavigationBar.tsx

```diff
@@ -0,0 +1,196 @@
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    "&:active": {
+      backgroundColor: "#0062BA"
+    },
+    [theme.breakpoints.down(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  buttonsMobile: {
+    marginBottom: '6px',
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    "&:active": {
+      backgroundColor: "#0062BA"
+    },
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  selectedBtn: {
+    float: 'left', 
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.down(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  selectedBtnMobile: {
+    marginBottom: '6px',
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  navMobile: {
+    border: '2px solid #008FD6',
+    '&:hover': {
+      border: '3px solid #008FD6'
+    },
+    borderRadius: '40px',
+    marginRight: '20px',
+    backgroundColor: "#022f51",
+    [theme.breakpoints.up(600)]: {
+      float: 'none',
+      display: 'none'
+    }
+  },
+  btnText: {
+    display: 'block', 
+    textAlign: "center", 
+    padding: '12px', 
+    borderBottom: '0',
+    "&:hover": {
+      borderBottom: "0px solid rgba(0,0,0,0)"
+    }
+  },
+  btnTextSelected: {
+    display: 'block',   
+    textAlign: "center", 
+    padding: '12px', 
+    borderBottom: '0',
+    color: 'white!important',
+    "&:hover": {
+      borderBottom: "0px solid rgba(0,0,0,0)"
+    },
+    width: '100%',
+    fontSize: '1.5rem'
+  }
+});
+
+export interface INavigationBar {
+  id: string;
+  showBackButton: boolean;
+  textResourceBindings: any;
+  triggers?: Triggers[];
+}
+
+export function NavigationBar(props: INavigationBar) {
+  const classes = useStyles();
+  const dispatch = useDispatch();
+  const orderedLayoutKeys = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.layoutOrder);
+  const returnToView = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.returnToView);
+  const pageTriggers = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.pageTriggers);
+  const triggers = props.triggers || pageTriggers;
+  const textResources = useSelector((state: IRuntimeState) => state.textResources.resources);
+  const currentView: string = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.currentView);
+  const showMenu: boolean = useSelector((state: IRuntimeState) => state.formLayout.uiConfig.showMenu);
```

Comment:
Does the expanded menu need to be exposed in the global state or can it be a simple useSate-hook? Is his state important to the rest of the app for some reason?

Reply:
I'm quite new to react so I'm not sure, I just saw how some of the similar functionality was implemented

Reply:
If you only need a state within the scope of a component you should use `useState(<your state value>)`. This method returns a tuple array with the first element containing the state and, the second element, a setter for the state.
In this case if nothing outside of the scope of the component needs to know about the visibility state of the menu I suggest replacing this with:
```js
const [showMenu, setShowMenu] = useState( false );
```
Then you can handle the state simply by using the `setShowMenu` function.
If any child component needs to be able to for example hide the menu, simply pass the `setShowMenu` function into the props of that component.